x-pack/plugin/core/src/main/java/org/elasticsearch/datastreams/logsdb/LogsdbIndexModeSettingsProvider.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.datastreams.logsdb;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.cluster.metadata.Metadata;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.compress.CompressedXContent;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.IndexMode;
+import org.elasticsearch.index.IndexSettingProvider;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.List;
+
+public class LogsdbIndexModeSettingsProvider implements IndexSettingProvider {
+    private static final Logger logger = LogManager.getLogger(LogsdbIndexModeSettingsProvider.class);
+    private static final PatternMatcher logsNameMatcher = PatternMatcher.forLogs();
+
+    public static final Setting<Boolean> CLUSTER_INDEX_MODE_LOGSDB_ENABLED = Setting.boolSetting(
+        "cluster.index.mode.logsdb.enabled",
+        false,
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+    private boolean isLogsdbEnabled;
+
+    public LogsdbIndexModeSettingsProvider(final ClusterService clusterService) {
```

Comment:
Let's remove the `ClusterService` dependency from this class? This would make unit testing this class easier.
We can do `clusterService.getClusterSettings().addSettingsUpdateConsumer(...)` from the plugin class. The setter is already defined here.